opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -1062,6 +1065,12 @@ def __init__(
         self.sampler = sampler
         self._span_limits = span_limits or SpanLimits()
         self._atexit_handler = None
+
+        self._resource._attributes = BoundedAttributes(
+            self._span_limits.max_attributes,
+            self._resource._attributes,
+            max_value_len=self._span_limits.max_attribute_length,
```

Comment:
This follows the existing pattern used with span events but technically mutates the resource as it'll end up truncating or dropping pre-existing attributes on a resource. I think in very strict terms it does mutate the resource but in practice this is what users would expect IMO. Curious to hear what other think about this.

Reply:
I kins of confused why span limits are used with resource. We don't want to mutate resource since it is prohibited by the spec. Did the spec introduce limits for Resources? that would be contradictory.

Reply:
Spec doesn't say anything about it AFAIK and I agree we can probably exclude resources completely. In practice the issues that limits solve don't really plague resources anyway. I'll remove this.

Reply:
I personally don't find using span limits for resources to be good. Can we take this up in separate issue?

Reply:
I agree with you. Let's discuss in the SIG call.